### ISSUE TITLE
Document the `adapterOptions` property on DS.Snapshot

### ISSUE THREAD

**Issue by bmac** [emberjs/data#3463]
https://github.com/emberjs/data/blob/4db3f1f949cca93a0af34996da7724d0a3364c41/packages/ember-data/lib/system/store.js#L651

**Comment by edwinallenz** [comment]
I would like to document the adapterOptions property, but it's hard for me to understand the reason for it and how it works, I you can help me to point me in the right direction I can do it definitely.

**Comment by wecc** [comment]
@edwinallenz Great, thanks!

IIRC you can pass `{ adapterOptions: { ... } }` as `options` in four different methods:

- `store.findAll(type, options)` - [source](https://github.com/emberjs/data/blob/4db3f1f949cca93a0af34996da7724d0a3364c41/packages/ember-data/lib/system/store.js#L1136)
- `store.findRecord(type, id, options)` - [source](https://github.com/emberjs/data/blob/4db3f1f949cca93a0af34996da7724d0a3364c41/packages/ember-data/lib/system/store.js#L611)
- `record.save(options)` - [source](https://github.com/emberjs/data/blob/4db3f1f949cca93a0af34996da7724d0a3364c41/packages/ember-data/lib/system/model/model.js#L737)
- `record.destroyRecord(options)` - [source](https://github.com/emberjs/data/blob/4db3f1f949cca93a0af34996da7724d0a3364c41/packages/ember-data/lib/system/model/model.js#L578)

We should document `options.adapterOptions` in those methods.

In addition to that, we should document the `adapterOptions` property on the `Snapshot` itself, something like [this](https://github.com/emberjs/data/blob/4db3f1f949cca93a0af34996da7724d0a3364c41/packages/ember-data/lib/system/snapshot.js#L90-L96).

Also, I think the line @bmac linked to should be updated to just pass the options to `internalModel.createSnapshot(options)` instead of setting them manually, like we do [here](https://github.com/emberjs/data/blob/4db3f1f949cca93a0af34996da7724d0a3364c41/packages/ember-data/lib/system/store.js#L1482).

**Comment by fidlip** [comment]
When you use store.findAll method, options parameter has no effect. It seems this parameter is not passed to (eg.) RESTAdapter, it's broken in RESTAdapter's findAll method, which has no snapshotArray parameter, which is expected to have. Declaration:
```
    findAll: function (store, type, sinceToken) { 
        ...
    }
```
but called with:
```
    ...
    var snapshotArray = recordArray.createSnapshot(options);
    var promise = adapter.findAll(store, typeClass, sinceToken, snapshotArray);
    ...
```
Or am I wrong?

**Comment by wecc** [comment]
@fidlip looks like it was fixed in #3850

**Comment by acorncom** [comment]
Looks like this issue can be closed?
